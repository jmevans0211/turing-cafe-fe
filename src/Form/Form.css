form {
  height: 25px;
  margin-bottom: 50px;
}

input, button {
  margin: 0 15px 20px 15px;
  height: 100%;
  width: 150px
}

button {
  background-color: teal;
  color: white;
}