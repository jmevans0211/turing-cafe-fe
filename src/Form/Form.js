import React, { Component } from 'react';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      customer: '',
      date: '',
      time: '',
      guests: ''
    }
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value})
     }

  handleSubmit = e => {
    e.preventDefault()
    this.props.addReservation(this.state)
  }

  render() {
    return (
      <form>
        <input placeholder="Name" type="text" name="customer" onChange={this.handleChange}/>
        <input placeholder="Date" type="text" name="date" onChange={this.handleChange}/>
        <input placeholder="Time" type="text" name="time" onChange={this.handleChange}/>
        <input placeholder="Number of Guests" type="number" name="guests" onChange={this.handleChange}/>
        <button onClick={this.handleSubmit}>Make Reservation</button>
      </form>
    )
  }
}

export default Form;