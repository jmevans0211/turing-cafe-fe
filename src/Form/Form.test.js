import React from 'react';
import { shallow } from 'enzyme'
import Form from './Form';


describe('Form', () => {
  let wrapper
  const mockAddReservation = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Form addReservation={mockAddReservation} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});



// describe('Form', () => {
//   it('should update name when handleChange is being called', () => {
//     // Setup
//     const mockFn = jest.fn();
//     const wrapper = shallow(<Form addReservation={mockFn}/>);
//     const mockEvent = { 
//       target: { 
//         name: 'name', 
//         value: 'Travis'
//       }
//     };
//     const expected = 'Travis';

//     // Expectation
//     expect(wrapper.state('name')).toEqual('');

//     // Execution
//     wrapper.instance().handleChange(mockEvent);

//     // Expectation
//     expect(wrapper.state('ideas')).toEqual(expected);
//   });
// });