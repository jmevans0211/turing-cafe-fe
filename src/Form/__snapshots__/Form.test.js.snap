// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date"
    type="text"
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="number"
  />
  <button
    onClick={[Function]}
  >
    Make Reservation
  </button>
</form>
`;
